fedimint-atomic-broadcast/src/keychain.rs

```diff
@@ -0,0 +1,103 @@
+use std::collections::BTreeMap;
+
+use aleph_bft::Keychain as KeychainTrait;
+use fedimint_core::PeerId;
+use secp256k1::hashes::sha256;
+use secp256k1::{schnorr, Message};
+
+use crate::conversion;
+
+#[derive(Clone, Debug)]
+pub struct Keychain {
+    pub peer_id: usize,
+    pub peer_count: usize,
+    pub threshold: usize,
+    public_keys: BTreeMap<PeerId, secp256k1::XOnlyPublicKey>,
+    keypair: secp256k1::KeyPair,
+    secp: secp256k1::Secp256k1<secp256k1::All>,
+}
+
+impl Keychain {
+    pub fn new(
+        peer_id: fedimint_core::PeerId,
+        public_keys: BTreeMap<PeerId, secp256k1::XOnlyPublicKey>,
+        secret_key: secp256k1::SecretKey,
+    ) -> Self {
+        let peer_id = peer_id.to_usize();
+        let peer_count = public_keys.len();
+        let threshold = 2 * peer_count / 3 + 1;
+        let secp = secp256k1::Secp256k1::new();
+        let keypair = secret_key.keypair(&secp);
+
+        Keychain {
+            peer_id,
+            peer_count,
+            threshold,
+            public_keys,
+            keypair,
+            secp,
+        }
+    }
+}
+
+impl aleph_bft::Index for Keychain {
```

Comment:
This seems weird. Maybe there are some reason for it, but "keychain being an index" is weird. Seems like "peer_id is an index" which makes sense, and the fact that keychain has an index is a implementation detail and IMO some abstraction layer is violated here.

Reply:
The aleph bft API requires its keychain to implement this trait.